### PR TITLE
Set build and e2e test workflows to test-and-builds environment

### PR DESCRIPTION
**Pull request by bradbown**
This PR implements issue(s) #

### Checklist

* [ ] I wrote new tests for my new core changes.
* [ ] I have successfully ran tests, style checker and build against my new changes locally.
* [ ] I have informed the team of any breaking changes if there are any.

### CODE REVIEW

File: .github/workflows/coordinator-build-and-publish.yml [hunk 1]
```diff
@@ -31,6 +31,7 @@ jobs:
   build-and-publish:
     runs-on: ubuntu-22.04
     name: Coordinator build
+    environment: test-and-builds
```
Comment (jpnovais) [inline .github/workflows/coordinator-build-and-publish.yml]: suggestion to the name: `docker-build-and-e2e`

test-and-builds - suggests that is for the unit test/inutial compilation/build

Comment (bradbown) [reply .github/workflows/coordinator-build-and-publish.yml]: Done

File: .github/workflows/coordinator-build-and-publish.yml [hunk 2]
```diff
@@ -31,6 +31,7 @@ jobs:
   build-and-publish:
     runs-on: ubuntu-22.04
     name: Coordinator build
+    environment: docker-build-and-e2e
```
Comment (Filter94) [inline .github/workflows/coordinator-build-and-publish.yml:38]: What is `docker-build-and-e2e` environment?

Comment (bradbown) [reply .github/workflows/coordinator-build-and-publish.yml:38]: This is github environment we've set up which the build and e2e test jobs have been added to. This stops the automatic running of these jobs unless we approve it.

**Review by Filter94 — APPROVED** [review]

**Review by VGau — APPROVED** [review]

**Review by jonesho — APPROVED** [review]

**Pull request merged**